src/t8_schemes/t8_new_feature/t8_subelements/t8_subelements_quad_cxx.cxx

```diff
@@ -0,0 +1,1325 @@
+  t8_quad_with_subelements *pquad_w_sub_sibling = (t8_quad_with_subelements *) sibling;
+
+  const p4est_quadrant_t *q = &pquad_w_sub_elem->p4q;
+  p4est_quadrant_t   *r = &pquad_w_sub_sibling->p4q;
+
+  T8_ASSERT (t8_element_is_valid (elem));
+  T8_ASSERT (t8_element_is_valid (sibling));
+
+  p4est_quadrant_sibling (q, r, sibid);
+  t8_element_copy_surround (q, r);
+}
+
+int
+t8_default_scheme_sub_c::t8_element_num_faces (const t8_element_t * elem)
+{
+  T8_ASSERT (t8_element_is_valid (elem));
+  return P4EST_FACES;
+}
+
+int
+t8_default_scheme_sub_c::t8_element_max_num_faces (const t8_element_t * elem)
+{
+  return P4EST_FACES;
+}
+
+int
+t8_default_scheme_sub_c::t8_element_num_children (const t8_element_t * elem)
+{ 
+  T8_ASSERT (t8_element_is_valid (elem));
+  return P4EST_CHILDREN;
+}
+
+int
+t8_default_scheme_sub_c::t8_element_num_face_children (const t8_element_t *
+                                                        elem, int face)
+{
+  T8_ASSERT (t8_element_is_valid (elem));
+  return 2;
+}
+
+int
+t8_default_scheme_sub_c::t8_element_get_face_corner (const t8_element_t *
+                                                      element, int face,
+                                                      int corner)
+{
+  /*
+   *   2    f_2    3
+   *     x -->-- x
+   *     |       |
+   *     ^       ^
+   * f_0 |       | f_1
+   *     x -->-- x
+   *   0    f_3    1
+   */
+
+  T8_ASSERT (t8_element_is_valid (element));
+  T8_ASSERT (0 <= face && face < P4EST_FACES);
+  T8_ASSERT (0 <= corner && corner < 2);
+  return p4est_face_corners[face][corner];
+}
+
+int
+t8_default_scheme_sub_c::t8_element_get_corner_face (const t8_element_t *
+                                                      element, int corner,
+                                                      int face)
+{
+  T8_ASSERT (t8_element_is_valid (element));
+  T8_ASSERT (0 <= corner && corner < P4EST_CHILDREN);
+  T8_ASSERT (0 <= face && face < 2);
+  return p4est_corner_faces[corner][face];
+}
+
+void
+t8_default_scheme_sub_c::t8_element_child (const t8_element_t * elem,
+                                            int childid, t8_element_t * child)
+{
+  const t8_quad_with_subelements *pquad_w_sub_elem = (const t8_quad_with_subelements *) elem;
+  t8_quad_with_subelements *pquad_w_sub_child = (t8_quad_with_subelements *) child;
+
+  const p4est_quadrant_t *q = &pquad_w_sub_elem->p4q;
+  p4est_quadrant_t   *r = &pquad_w_sub_child->p4q;
+
+  const p4est_qcoord_t shift = P4EST_QUADRANT_LEN (q->level + 1);
+
+  T8_ASSERT (t8_element_is_valid (elem));
+  T8_ASSERT (t8_element_is_valid (child));
+  T8_ASSERT (p4est_quadrant_is_extended (q));
+  T8_ASSERT (q->level < P4EST_QMAXLEVEL);
+  T8_ASSERT (childid >= 0 && childid < P4EST_CHILDREN);
+  
+  r->x = childid & 0x01 ? (q->x | shift) : q->x;
+  r->y = childid & 0x02 ? (q->y | shift) : q->y;
+  r->level = q->level + 1;
+  
+  T8_ASSERT (p4est_quadrant_is_parent (q, r));
+
+  t8_element_copy_surround (q, r);
```

Comment:
I inserted an assertion in case of subelements for now. I dont think we need to set the subelement infos here since the child element is already initialized with default subelement infos at this point and it will never get non-default subelement infos since this function will not be used by subelements.